### PR TITLE
Anonymous tasks

### PR DESCRIPTION
**Pull request by dorukozturk**
This PR adds necessary changes to run jobs as a non-authenticated user.
When a user is authenticated the uploaded files and job results will be uploaded to user's private folder.
When a user is not authenticated, there will be an anonymous user that will be logged in and uploaded files and job results will be uploaded to anonymous user's private folder.

### CODE REVIEW

**Comment by jeffbaumes** [comment]
I really like this solution: just make a user with a known insecure password and login as that user using the normal login flow. No new endpoints - elegant. I'll test it out and approve once it's working for me.

File: client/src/App.vue [hunk 1]
```diff
@@ -12,21 +12,25 @@
...
-  }),
-  asyncComputed: {
-    async me() {
-      return (await this.girderRest.get('user/me')).data;
-    },
-    async scratchFolder() {
-      const me = (await this.girderRest.get('user/me')).data;
-      const folders = (await this.girderRest.get(`folder?parentId=${me._id}&parentType=user`)).data;
-      return folders.filter(folder => folder.name === 'Private')[0];
+    name: 'app',
```
Comment (jeffbaumes) [inline client/src/App.vue]: Please update this to two-space indentation as in the rest of the JS

**Review by jeffbaumes — CHANGES_REQUESTED** [review]
This worked perfectly for me. Just a small code style change suggested.

**Comment by dorukozturk** [comment]
@jeffbaumes I fixed the indentation. By the way Dan came up with the idea which I also thought made more sense than the separate endpoint. I am glad it worked for you.

**Comment by jeffbaumes** [comment]
There is a slight potential issue, where anyone could go in and change the anon password and mess things up, where a special endpoint that doesn't even care about a password could avoid that case. I'd easily take the code simplicity here over the added complexity of a new endpoint, so I'm good with this solution.

**Review by jeffbaumes — APPROVED** [review]

**Pull request merged**